packages/ra-ui-materialui/src/input/BooleanInput.stories.tsx

```diff
@@ -16,6 +16,18 @@ export const Basic = () => (
     </Wrapper>
 );
 
+type Post = {
+    published: boolean;
+    reported: boolean;
+    title: string;
+};
+
+export const Typed = () => (
+    <Wrapper>
+        <BooleanInput<Post> source="published" />
```

Comment:
If I remember correctly, the `source` prop only allowed the value `"published"` or `"reported"` with the previous version. But now all 3 values (`"published"`, `"reported"`, `"title"`) are possible. Is this regression intended.

Reply:
Same as above